app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -19,6 +19,10 @@ def handle_second_factor_locked_user
   end
 
   def check_already_authenticated
+    unless user_session
+      redirect_to new_user_session_path
```

Comment:
Yeah we should test all controllers. Since there doesn't appear to be a shared behavior already for the things that include this, and shared behaviors can be hard to track down and read, how do you feel about me copy-pasting this block to the other controller specs?